src/lib/homebrew/homebrew.js

```diff
@@ -154,6 +158,26 @@ class Homebrew {
         const setupSystemTiles = new SetupSystemTiles();
         setupSystemTiles.clean();
         setupSystemTiles.setup();
+
+        return this;
+    }
+
+    resetGenericPromissoryNotes() {
+        console.log("Homebrew.resetGenericPromissoryNotes");
+
+        if (world.TI4.config.timestamp > 0) {
+            Broadcast.chatAll(
+                locale("ui.error.homebrew.resetAfterLoaded"),
+                Broadcast.ERROR
+            );
+            return;
+        }
+
+        const setupGenericPromissory = new SetupGenericPromissory();
```

Comment:
So, I thought this should be:
```
        for (const playerDesk of world.TI4.getAllPlayerDesks()) {
            const setupGenericPromissory = new SetupGenericPromissory(playerDesk);
            setupGenericPromissory.clean();
            setupGenericPromissory.setup();
        }
```
But in trying that out, my TTPG freezes up completely.